lib/sitemap_generator/builder/sitemap_file.rb

```diff
@@ -67,7 +67,7 @@ def empty?
       # bytesize will be calculated for you.
       def file_can_fit?(bytes)
         bytes = bytes.is_a?(String) ? SitemapGenerator::Utilities.bytesize(bytes) : bytes
-        (@filesize + bytes) < SitemapGenerator::MAX_SITEMAP_FILESIZE && @link_count < SitemapGenerator::MAX_SITEMAP_LINKS && @news_count < SitemapGenerator::MAX_SITEMAP_NEWS
+        (@filesize + bytes) < SitemapGenerator::MAX_SITEMAP_FILESIZE && @link_count < SitemapGenerator::Sitemap.max_sitemap_links && @news_count < SitemapGenerator::MAX_SITEMAP_NEWS
```

Comment:
As mentioned below, pass in a `:max_sitemap_links` option to the initializer, assign it to `@max_sitemap_links` instance variable and reference that here.